packages/grafana-ui/src/components/Table/HeaderRow.tsx

```diff
@@ -58,6 +60,7 @@ function renderHeaderCell(column: any, tableStyles: TableStyles, field?: Field)
             className={tableStyles.headerCellLabel}
             title={column.render('Header')}
           >
+            {showTypeIcons && <Icon name={getFieldTypeIcon(field?.type)} style={{ marginRight: '8px' }} />}
```

Comment:
size="sm" could work better (be more similar to text size). And would be nice to have add a title="type name" , so you can hover over the icon and get the type name